brave/src/main/java/com/linecorp/armeria/client/brave/ArmeriaHttpClientParser.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.brave;
+
+import java.net.SocketAddress;
+
+import com.linecorp.armeria.common.RpcRequest;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.internal.brave.SpanTags;
+
+import brave.SpanCustomizer;
+import brave.http.HttpAdapter;
+import brave.http.HttpClientParser;
+
+/**
+ * Default implementation of {@link HttpClientParser}.
+ * This parser add some custom tags and overwrite the name of span if {@link RequestLog#requestContent()}
+ * is {@link RpcRequest}.
+ * The following tags become available:
+ * <ul>
+ *   <li>http.url</li>
+ *   <li>http.host</li>
+ *   <li>http.protocol</li>
+ *   <li>http.serfmt</li>
+ *   <li>address.remote</li>
+ *   <li>address.local</li>
+ * </ul>
+ * User can extend this class or implement own {@link HttpClientParser}.
+ */
+public class ArmeriaHttpClientParser extends HttpClientParser {
```

Comment:
Ditto - Does it make sense to make this class a singleton and let a user get it via `ArmeriaHttpClientParser.get()`?